docs/faqs/contributor-faq.md

```diff
@@ -0,0 +1,218 @@
+
+Each time an uncaught exception occurs, the `OPENMS_DUMP_CORE` variable is checked and a segmentation fault is caused, if it is set.
+
+### (Linux) Why is no core dumped, although a fatal error occured?
+
+Try the `ulimit -c` unlimited command. It sets the maximum size of a core to unlimited.
+
+> **_NOTE:_**  We observed that, on some systems, no core is dumped even if the size of the core file is set to unlimited. We are not sure what causes this problem
+
+### (Linux) How can I set breakpoints in gdb to debug OpenMS?
+
+Imagine you want to debug the TOPPView application and you want it to stop at line 341 of SpectrumMDIWindow.C.
+
+1. Enter the following in your terminal:
+
+  ```bash
+  Run gdb:
+ shell> gdb TOPPView
+```
+
+2. Start the application (and close it):
+
+  ```bash
+ gdb> run [arguments]
+```
+3. Set the breakpoint:
+  ```bash
+ gdb> break SpectrumMDIWindow.C:341
+```
+4. Start the application again (with the same arguments):
+
+  ```bash
+ gdb> run
+ ```
+
+## Cross-platform thoughts
+
+OpenMS runs on three major platforms. Here are the most prominent causes of "it runs on Platform A, but not on B. What now?"
+
+### Reading or writing binary files
+Reading or writing binary files causes different behaviour. Usually Linux does not make a difference between text-mode and binary-mode when reading files. This is quite different on Windows as some bytes are interpreted as `EOF`, which lead might to a premature end of the reading process.
+
+If reading binary files, make sure that you explicitly state that the file is binary when opening it.
+
+During writing in text-mode on Windows a line-break (`\n`) is expanded to (`\r\n`). Keep this in mind or use the `eol-style` property of subversion to ensure that line endings are correctly checked out on non-Windows systems.
+
+### `UInt` vs `Size`
+Both `unsigned int` vs `size_t` `UInt` and `Size` have the same size on Linux GCC (32bit on 32bit systems, 64bit on 64 bit systems), however on Windows this only holds for 32bit. On a 64bit Windows, the `UInt` type is still 32bit, while the `Size` type is 64bit. This might lead to warnings (at best) or overflows and other drawbacks.
+Therefore, do not assume that `UInt` is equal to `Size`.
+
+### Paths and system functions
+
+Avoid hardcoding e.g.`String tmp_dir = "/tmp";`. This will fail on Windows. Use Qt's `QDir` to get a path to the systems temporary directory if required.
+
+Avoid names like uname which are only available on Linux.
+
+When working with files or directories, it is usually safe to use "/" on all platforms. Take care of spaces in directory names though. You should always quote paths if they are used in a system call to ensure that the subsequent interpreter takes the spaced path as a single entity.
+
+## Doxygen Documentation
+
+### Where can I find the definition of the main page?
+
+You can find a definition of the main page [here](https://github.com/OpenMS/OpenMS/edit/develop/doc/doxygen/public/Main.doxygen).
```

Comment:
i'm not sure, if i'm repeating myself on this text again:

delete:

> You can.

(ofcourse, I can)